pilot/pkg/model/validation.go

```diff
@@ -1222,22 +1222,21 @@ func ValidateDestinationPolicy(msg proto.Message) error {
 
 // ValidateProxyAddress checks that a network address is well-formed
 func ValidateProxyAddress(hostAddr string) error {
-	colon := strings.Index(hostAddr, ":")
-	if colon < 0 {
-		return fmt.Errorf("':' separator not found in %q, host address must be of the form <DNS name>:<port> or <IP>:<port>",
-			hostAddr)
+	host, p, err := net.SplitHostPort(hostAddr)
+	if err != nil {
+		return fmt.Errorf("unable to split %q: %s", hostAddr, err.Error())
```

Comment:
nit: %v and err directly, here and below

Reply:
Thanks! Will update.